src/services/address.service.ts

```diff
@@ -14,15 +14,41 @@ class AddressService {
         return;
       }
 
-      this.request(addressRequest)
-        .then((response) => {
-          resolve({
-            count: response.size(),
+      if (addressRequest.body.page != undefined) {
+        this.request(addressRequest)
+          .then((response: Array<Object>) => {
+            resolve(
+              {
+                count: response.length
+              }
+            )
+          })
+          .catch((err) => {
+            reject(err);
           });
-        })
-        .catch((err) => {
-          reject(err);
+      }
+      else {
+        let count = 0;
+        let page = 1;
+        let responseSize = 0;
+        do {
```

Comment:
A do/while? Are you just fucking with me?

Reply:
An alternative to this is:

`let responseSize = 1000;`
`while(responseSize === 1000) { }`